### PR TITLE
Add version tags in txt files (cleaned version)

### PR DESCRIPTION
**Pull request by BaptisteCecconi**
This PR replaces the two previous PR by @loumir (#77 and #75)

### CODE REVIEW

File: convert_ucd_list_to_tex.py [hunk 1]
```diff
@@ -11,7 +11,8 @@ def ucd_to_tex(file):
     with open(file, 'r') as f:
         lines = f.readlines()
         for cur_line in lines:
-            convert_line(cur_line)
+            if not cur_line.startswith("#"):
+                 convert_line(cur_line)
```
Comment (msdemlei) [inline convert_ucd_list_to_tex.py:15]: The text above says "Comment rows start" -- I *think* you mean "comment lines", in which case this code would match the text.  The current text, however, sounds more like "delete all chars starting with a hash regardless of where it starts within a line".  We'll have to choose one or the other.

**Review by msdemlei — CHANGES_REQUESTED** [review]

File: UCDlist.tex [hunk 2]
```diff
@@ -760,7 +772,7 @@ \subsection{Remarks on combination rules}
 and included in the list of UCD words. See Section \ref{sec:list} with the tags definitions on top.
 
 They correspond to real usage of the terms in science publications and are attached to the description 
-of catalogues’ column by experimented data scientists. UCD combination also reflects the catalogues 
+of catalogues' column by experimented data scientists. UCD combination also reflects the catalogues
```
[the remark this answers is not in the document]
Comment (loumir) [reply UCDlist.tex:775]: yes correct with 'columns' and 'experienced'.

File: UCDlist.tex [hunk 3]
```diff
@@ -803,29 +815,38 @@ \subsection{P or S syntax code: Which is the most pertinent position for a UCD w
...
 
+\subsection{Version tags in file}
+
+The first row of each of these files is a comment line containing the UCDList version 
+tag following this syntax: \texttt{\$DOCTYPE-\$DOCNAME-v\$DOCVERSION-\$DOCDATE}, such as, 
+e.g., \texttt{PEN-UCDlist-v1.5-20220706}.
+
+Solely the lists with version starting by \texttt{EN-UCDList} are valid reference for UCD 
+terms or deprecated terms. They correspond to the Endorsed Note at the final step of the 
+update cycle. Others are only work-in-progress and should not be referenced.
```
[the remark this answers is not in the document]
Comment (loumir) [reply UCDlist.tex:848]: I suggest "should not be referenced by a url".  The text before explains the intent.

File: convert_ucd_list_to_tex.py (hunk 1 above)
Comment (loumir) [reply convert_ucd_list_to_tex.py:15]: ok for "Comment lines".

**Comment by msdemlei** [comment]
On Tue, Jul 26, 2022 at 02:13:59AM -0700, Mireille LOUYS wrote:
> @loumir commented on this pull request.
> +update cycle. Others are only work-in-progress and should not be referenced. 
> 
> I suggest "should not be referenced by a url".  The text before
> explains the intent. 

Hm...  The thing *has* a URI as soon as we push it to github, and
that URI will be at least on github's web pages, which certainly is a
reference.  And what should be wrong with that?  Hence, I don't think
we can sensibly write "should not be referenced" either.

I'd still say the only sensible thing we can say is "should not be
used in released software".  In which case the version tag might
indeed serve some purpose -- any UCD list that doesn't have -EN- in
it must not be out in the wild.

If we have to have that version tag: could you, Mireille, perhaps
contribute a test script that ensures that version tag is ok on make
test?
https://ivoa.net/documents/Notes/IVOATexDoc/20220525/NOTE-ivoatexDoc-1.3-20220525.html#tth_sEc3.11
has a few words on the various options for that.  As I said, without
machine supervision, there's no way this won't break...

File: convert_ucd_list_to_tex.py (hunk 1 above)
Comment (loumir) [reply convert_ucd_list_to_tex.py:15]: corrections are included in #PR 80

**Comment by msdemlei** [comment]
Since I can't push commits in here, I've moved all this to PR #81, where there's now the make test thing I was asking for.

**Pull request closed without merging**